src/app/robots.ts

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { MetadataRoute } from "next";
+
+export default function robots(): MetadataRoute.Robots {
+  return {
+    rules: {
+      userAgent: "*",
+      allow: "/",
+      disallow: ["/user/dashboard/", "/user/settings/", "/user/welcome/"],
```

Comment:
Question: Is it safer to disallow the paths explicitly or why not just do `/user/*`?

Reply:
Yeah I did consider that, but I was worried that I couldn't be confident that we wouldn't in the future put any public-facing pages under `/user/`, and then not realise that they wouldn't be crawled. That failure mode seemed worse to me than a loginwalled page inadvertently being indexed.